dist/init/linux-systemd/caddy.service

```diff
@@ -5,18 +5,32 @@ After=network-online.target
 Wants=network-online.target systemd-networkd-wait-online.service
 
 [Service]
+; Only allow 1 restart with 10 seconds so that caddy doesn't get locked out by PAM
 Restart=on-failure
-StartLimitInterval=86400
-StartLimitBurst=5
+StartLimitInterval=10
+StartLimitBurst=1
 
 ; User and group the process will run as.
 User=www-data
 Group=www-data
 
+; Allow ExecStartPre to run as root, but restrict ExecStart
+PermissionsStartOnly=true
+
+; Make sure that /etc/ssl/caddy exists with the correct permissions
+ExecStartPre=/bin/mkdir -p /etc/ssl/caddy
+ExecStartPre=/bin/chown -R www-data:www-data /etc/ssl/caddy
+ExecStartPre=/bin/chmod 750 /etc/ssl/caddy
```

Comment:
From `PermissionsStartOnly` to all `ExecStartPre`: Creating non-ephemeral prerequisite directories is the responsibility of the *pre-install* scripts of *packages*. Therefore we've put the instructions, as per tradition, in the `README` file. ;-)

As we're working here with **systemd** please remove these lines and create a file  
`/etc/tmpfiles.d/caddy.conf` with (incomplete):
```
# /etc/tmpfiles.d/caddy.conf

d /etc/ssl/caddy 0700 www-data www-data
d /etc/caddy …
```
… which a release engineer would've shipped in this case. *systemd* will then create the folders immediately—or, depending on your distribution's setup or/and its *systemd* version, after some time. In that case, run `systemd-tmpfiles --create` to see the result right away.

(Add the file, and where to copy it to the `README`.)